server/clip_server/executors/mclip_hg.py

```diff
@@ -0,0 +1,276 @@
+import os
+import warnings
+from multiprocessing.pool import ThreadPool
+from typing import Dict, Optional
+import numpy as np
+import torch
+from transformers import CLIPFeatureExtractor, CLIPModel, PreTrainedModel, PretrainedConfig, AutoTokenizer, AutoModel
+from clip_server.executors.helper import (
+    split_img_txt_da,
+    set_rank,
+)
+from jina import Executor, requests, DocumentArray
+
+
+class MCLIPConfig(PretrainedConfig):
+    model_type = "M-CLIP"
+
+    def __init__(self, modelBase='xlm-roberta-large', transformerDimSize=1024, imageDimSize=768, **kwargs):
+        self.transformerDimensions = transformerDimSize
+        self.numDims = imageDimSize
+        self.modelBase = modelBase
+        super().__init__(**kwargs)
+
+
+class MultilingualCLIP(PreTrainedModel):
+    config_class = MCLIPConfig
+
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(config, *args, **kwargs)
+        self.transformer = AutoModel.from_pretrained(config.modelBase)
+        self.LinearTransformation = torch.nn.Linear(in_features=config.transformerDimensions,
+                                                    out_features=config.numDims)
+
+    def forward(self, txt_tok):
+        embs = self.transformer(**txt_tok)[0]
+        att = txt_tok['attention_mask']
+        embs = (embs * att.unsqueeze(2)).sum(dim=1) / att.sum(dim=1)[:, None]
+        return self.LinearTransformation(embs)
+
+
+class CLIPEncoder(Executor):
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str = 'M-CLIP/XLM-Roberta-Large-Vit-B-32',
+        clip_model_name_or_path: str = 'openai/clip-vit-base-patch32',
```

Comment:
For the `M-CLIP` style models, is there a list of the variate models? I guess we need maintain a model list to tell the user how to chose the right model regarding to their use case.

Reply:
What's more, can we kill the `pretrain_model_name_or_path`, and share the `clip_model_name` parameter to specify which model to load?